src/jit/lowerxarch.cpp

```diff
@@ -2300,42 +2300,25 @@ void Lowering::ContainCheckSIMD(GenTreeSIMD* simdNode)
 //
 void Lowering::ContainCheckHWIntrinsic(GenTreeHWIntrinsic* node)
 {
-    NamedIntrinsic intrinsicID = node->gtHWIntrinsicId;
-    GenTree*       op1         = node->gtGetOp1();
-    GenTree*       op2         = node->gtGetOp2();
+    NamedIntrinsic      intrinsicID = node->gtHWIntrinsicId;
+    HWIntrinsicCategory category    = comp->categoryOfHWIntrinsic(intrinsicID);
+    int                 numArgs     = comp->numArgsOfHWIntrinsic(intrinsicID);
+    GenTree*            op1         = node->gtGetOp1();
+    GenTree*            op2         = node->gtGetOp2();
 
-    switch (node->gtHWIntrinsicId)
+    // TODO-XArch-CQ: Non-VEX encoded instructions can have both ops contained
+    // TODO-XArch-CQ: Non-VEX encoded instructions require memory ops to be aligned
+    if ((category & HW_Category_SimpleSIMD) != 0 && numArgs == 2 && comp->canUseVexEncoding())
```

Comment:
I don't think its clear what will qualify as `SimpleSIMD` and what won't.

`MoveHighToLow` for example, would appear to be a `SimpleSIMD` intrinsic, at first glance. But it can't be categorized as such since it can't take a mem operand.

Several of the `ival` intrinsics can also go through the same containment analysis here since the second op can be contained (the only difference is they pass a third op immediate).

Reply:
The current implementation only applies containment on 2-arg simple SIMD intrinsic. We can change here later?

Reply:
I agree we can change later.

However, I think it would be good to ensure we have the scenarios we already know need to be covered done before merging this PR.

That is, https://github.com/dotnet/coreclr/pull/15538 already implements basic codegen for the entire SSE Isa. It would be nice to have a clear decision on something like `NI_SSE_MoveHighToLow`, which qualifies as a `SimpleSIMD` intrinsic based on your definition [here](https://github.com/dotnet/coreclr/pull/15749/files#diff-f9f092d1e72ff73d11a88244d4f0c27aR29), but doesn't qualify for the containment check being done here, due to it not supporting a `[mem]` operand.

Reply:
Perhaps, we can define a new category `HW_Category_NoContainment` and define `NI_SSE_MoveHighToLow` as `HW_Category_SimpleSIMD | HW_Category_NoContainment`.

Reply:
Possibly (I still think it would be better as two distinct fields: `category` and `flags`, but either achieves the same thing in the end).